### PR TITLE
fix: refactor mtls logic to standalone method

### PR DESCRIPTION
**Pull request by arithmetic1728**
refactor the code to a standalone `get_mtls_endpoint_and_cert_source` class method, so the logic can be re-used by gapic clients with handwritten layers to fix the mtls in their handwritten layer.

### CODE REVIEW

**Comment by arithmetic1728** [comment]
integration test failure is due to #1122

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/async_client.py.j2 [hunk 1]
```diff
@@ -90,6 +90,40 @@ class {{ service.async_client_name }}:
...
+                in this method.
+
+        Returns:
+            Tuple[str, Callable[[], Tuple[bytes, bytes]]]: returns the API endpoint and the
+                client cert source to use.
+        
+        Raises:
+            google.auth.exceptions.MutualTLSChannelError: If any errors happen.
+        """
+        return {{ service.client_name }}.get_mtls_endpoint_and_cert_source.__func__({{ service.async_client_name }}, client_options)  # type: ignore
```
Comment (busunkim96) [inline gapic/templates/%namespace/%name_%version/%sub/services/%service/async_client.py.j2:125]: Why is `__func__` needed to call the function?

Comment (busunkim96) [inline gapic/templates/%namespace/%name_%version/%sub/services/%service/async_client.py.j2:125]: Is the __func__ necessary to call the function?

Comment (arithmetic1728) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/async_client.py.j2]: Yea it's not needed. I just changed it to `{{ service.client_name }}.get_mtls_endpoint_and_cert_source(client_options)`.

Comment (arithmetic1728) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/async_client.py.j2]: removed it in the new commit

**Review by busunkim96 — APPROVED** [review]

**Pull request merged**